cmd/login/wc.go

```diff
@@ -147,65 +200,90 @@ func (r *runner) createClusterClientCert(ctx context.Context, client k8sclient.I
 		certOperatorVersion: certOperatorVersion,
 	}
 
-	clientCertResource, err := createCert(ctx, clientCertService, certConfig)
+	clientCertResource, secret, err := r.getCredentials(ctx, services.clientCertService, certConfig)
 	if err != nil {
-		return microerror.Mask(err)
+		return "", false, microerror.Mask(err)
 	}
 
-	// Retrieve client certificate credential.
-	secret, err := fetchCredential(ctx, provider, clientCertService, clientCertResource)
+	credentialConfig := credentialConfig{
+		clusterID:     r.flag.WCName,
+		certCRT:       secret.Data[credentialKeyCertCRT],
+		certKey:       secret.Data[credentialKeyCertKey],
+		certCA:        secret.Data[credentialKeyCertCA],
+		clusterServer: getServerAddress(certConfig),
+		filePath:      r.flag.SelfContained,
+		loginOptions:  r.loginOptions,
+	}
+
+	contextName, contextExists, err := r.storeWCCredentials(credentialConfig)
 	if err != nil {
-		return microerror.Mask(err)
+		return "", false, microerror.Mask(err)
 	}
 
-	// Store client certificate credential either into the current kubeconfig or a self-contained file if a path is given.
-	var contextExists bool
-	var contextName string
-	if r.loginOptions.selfContainedWC {
-		contextName, contextExists, err = printWCCredentials(r.k8sConfigAccess, r.fs, r.flag.SelfContained, clientCertResource, secret, clusterBasePath, r.loginOptions)
+	if clientCertResource != nil {
```

Comment:
We only clean up if a clientCertResource has been created (non openstack case)

Reply:
Could you add what you said above as a comment in the code?